src/parser/mod.rs

```diff
@@ -3426,6 +3426,27 @@ impl<'a> Parser<'a> {
         Ok(values)
     }
 
+    fn parse_comma_separated_end(&mut self) -> Option<Token> {
```

Comment:
Seems the returned token is never used, so it should be good to use `bool` instead of `Option`? @iffyio How do you think?

Reply:
Upon re-reading the implementation, the return value is confusing (it returns `Some(Token::Comma)` if there is no comma and the method name is confusing as well. Maybe it should just be `fn is_comma_separated_end(&self) -> bool` and the call-site is `if !self.consume_token(&Token::Comma) || self.is_comma_separated_end() { break; }`

Reply:
Yes, but I'm wondering if it'll be strange to return `ControlFlow` since the function mainly checks if it reached the end. I prefer changing to `is_parse_comma_separated_end` and returning a boolean value.

Reply:
Yeah calling the function `is_parse_comma_separated_end` and having it return a boolean sound reasonable to me!